cli/commands/project_config.go

```diff
@@ -35,33 +36,48 @@ type ProjectConfig struct {
 
 // Create a new ProjectConfig using configured or default locations
 func NewProjectConfig() *ProjectConfig {
+	readyConfig := &ProjectConfig{}
 	projectConfigFile := os.Getenv("RIG_PROJECT_CONFIG_FILE")
 
-	var discovery []string
 	if projectConfigFile == "" {
-		discovery = make([]string, 2)
-		discovery[0] = "./outrigger.yml"
-		discovery[1] = "./.outrigger.yml"
-	} else {
-		discovery = make([]string, 1)
-		discovery[0] = projectConfigFile
+		projectConfigFile, _ = FindProjectConfigFilePath()
 	}
 
-	readyConfig := &ProjectConfig{}
-	for _, filePath := range discovery {
-		if config, err := NewProjectConfigFromFile(filePath); err == nil {
+	if projectConfigFile != "" {
+		if config, err := NewProjectConfigFromFile(projectConfigFile); err == nil {
 			readyConfig = config
-			break
 		}
 	}
 
 	return readyConfig
 }
 
-// Create a new ProjectConfig from the specified file
+// Traverse directory structure looking for an outrigger project config file.
+func FindProjectConfigFilePath() (string, error) {
```

Comment:
If I understand the code correctly, this could ascend up and out of the project directory and pick up an outrigger file that may be lingering around in a directory above where the project is checked out. Seems like an unlikely edge case (should only happen if someone is trying to use a rig command in a project that isn't set up for it). I'm not sure an effective way to prevent against it (maybe don't ascend above any directory with a .git directory in it) but should we document somewhere that this is possible?

In thinking through whether we could make a recommendation to have an outrigger.yml file at the user directory level or their "Project" directory to catch when this happens and alert people I can't think of a mechanism where that wouldn't just cause confusion as there isn't a way to put in a global echo command I can think of.

Perhaps just a verbose logging of what outrigger file is being used is the path isn't os.Getwd()?

Reply:
I see that the logging is down below in the unison sync process setup and above in the project scripts command. I had missed it before so it looks like this is already handled.